### ISSUE TITLE
Performance comparison with original paper

### ISSUE THREAD

**Issue by bwang482** [30stomercury/Interaction-aware_Attention_Network#2]
Thanks very much for the implementation @30stomercury ! It is great!

Am I right that currently there is no evaluation on a validation set during training, thus why no early stopping is used for training? There is a fixed training steps (e.g. 2000 steps). 

Have you managed to replicate the results reported in the original paper? I have only been able to get 55.4 in UAR, which is far lower than what they have reported:

```
[[115  26  26   3]
 [ 74 149 172  47]
 [ 27  31 255  71]
 [ 15  36  49 145]]
Ave test acc: 0.535, Ave test uar: 0.567
ave uar over all data: 0.5547344560992651
```

Thanks again!

**Comment by 30stomercury** [comment]
Hi!
1. I've added input parameter based on the  setup in section 3.1. ```-lr 0.0001 -batch_size 64 -keep_proba 0.9 -seq_dim 512 -save_path ./model/iaan/```; moreover, for convenience you can find the extracted feature `feat_pooled.pkl` and original dataset from the link in ./data/ folder. I thought you haven't setup the optimized parameters the paper reported.
2. Since there is no evaluation on a validation set, I adopt leave-one-session-out cross validation, which is commonly used for IEMOCAP evaluation.
3. I suggest you to extend the training steps to 3000 steps. For checkpoints selection, you can either pick the checkpoint with highest uar of each session in 3000 steps or stop the training process if there are no improvement in successive 5 evaluation steps on test session, then pick the checkpoints with  highest uar as well. I am able to replicate the results, although the optimized checkpoints may not be the same for each session every training time, I got similar results on LOSO cross validation.

Thank you @bluemonk482 for trying to replicate this work.

**Comment by bwang482** [comment]
Thanks very much @30stomercury ! That's very useful!

I don't seem to have the permission to download the features from this [link](https://drive.google.com/drive/u/1/folders/1rJk4V5YeQNTOtT0WjZ-VlEzWg4c6-q1R). But I think I have generated the features in the correct way. After training and picking the checkpoint with the highest UA, my final results are 62.36 for WA, and 64.36 for UA, which is now close to the results reported in the paper. 

Is there anything I can do to give it another push to get 64+ and 66+ for WA and UA? I am just wondering if I have missed anything. Or do the results vary (+-2) every time I run the experiments? Thanks again!

**Comment by 30stomercury** [comment]
@bluemonk482 really sorry for the late reply, I'm working on another project heavily these days.
I've thoroughly check the original version, I didn't remove frames with F0=0 during preprocessing in  line 75, 76 in data/preprocess.py. Regarding to the performance, the results should vary within 1 every time.

Thanks for reporting your results to me, it's really valuable.
Song lin

**Comment by bwang482** [comment]
Thanks very much @30stomercury ! You're very helpful! I will close the issue now.
